### PR TITLE
[DYN-3027] Libraries CoreNodeModels coverage

### PR DESCRIPTION
**Pull request by Astul-Betizagasti**
### Purpose

Increase Libraries/CoreNodeModes test coverage

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers

Aaron Tang (@QilongTang )
Michael Kirschner (@mjkkirschner )
Aparajit Pratap (@aparajit-pratap )

### FYIs

Alfredo Pozo (@alfredo-pozo )

### CODE REVIEW

File: test/Libraries/CoreNodesTests/CoreNodeModelsTests/SelectionTests.cs [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+        private NodeModel testNode;
+        private Mock<IModelSelectionHelper<ModelBase>> selectionHelperMock;
+
+        [SetUp]
+        public void TestsSetup()
+        {
+            selectionHelperMock = new Mock<IModelSelectionHelper<ModelBase>>(MockBehavior.Strict);
+            selection = new SelectionConcrete(SelectionType.Many, SelectionObjectType.Element, "testMessage", "testPrefix", selectionHelperMock.Object);
+            selection.Name = "selectionTestName";
+            testNode = new DummyNode();
```
Comment (aparajit-pratap) [inline test/Libraries/CoreNodesTests/CoreNodeModelsTests/SelectionTests.cs:29]: Can we test this with some other node? I think you used `CodeBlockNode` before. Will that work now? Selecting a DummyNode may not be all that useful to test.

Comment (Astul-Betizagasti) [reply test/Libraries/CoreNodesTests/CoreNodeModelsTests/SelectionTests.cs:29]: Yes it should work, I'll change it and update the pr

File: test/TestUINodes/SelectionConcrete.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            string prefix)
+                : base(
+                    selectionType,
+                    selectionObjectType,
+                    message,
+                    prefix)
+        {
+        }
+
+        public SelectionConcrete(
```
Comment (aparajit-pratap) [inline test/TestUINodes/SelectionConcrete.cs:75]: Do we need this constructor? Is it used?

**Review by aparajit-pratap — APPROVED** [review]

File: test/TestUINodes/SelectionConcrete.cs (hunk 2 above)
Comment (QilongTang) [reply test/TestUINodes/SelectionConcrete.cs:75]: Seems this one is used in `TestsSetup`

**Review by QilongTang — APPROVED** [review]

File: test/TestUINodes/SelectionConcrete.cs (hunk 2 above)
Comment (Astul-Betizagasti) [reply test/TestUINodes/SelectionConcrete.cs:75]: Yes, that constructor is not being used, the one Aaron mentioned is the one on line 47. I will remove it

Comment (QilongTang) [reply test/TestUINodes/SelectionConcrete.cs]: ah thanks!

**Pull request merged**